flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/ConsumedPartitionGroup.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.flink.runtime.scheduler.strategy;
+
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+
+import java.util.Collections;
+import java.util.List;
+
+/** Group of consumed {@link IntermediateResultPartitionID}s. */
+public class ConsumedPartitionGroup {
+    private final List<IntermediateResultPartitionID> resultPartitions;
+
+    public ConsumedPartitionGroup(List<IntermediateResultPartitionID> resultPartitions) {
+        this.resultPartitions = resultPartitions;
+    }
+
+    public ConsumedPartitionGroup(IntermediateResultPartitionID resultPartition) {
+        this(Collections.singletonList(resultPartition));
+    }
+
+    public List<IntermediateResultPartitionID> getResultPartitions() {
+        return Collections.unmodifiableList(resultPartitions);
+    }
```

Comment:
We could hide the implementation detail by letting `ConsumedPartitionGroup` implement the methods we need it to have to directly work with it. For example if it implements `size()` and `Iterable`, then it should already go a far way. Maybe we also need `get(int index)`. The same applies to the `ConsumerVertexGroup`.